src/halmos/solvers.py

```diff
@@ -0,0 +1,400 @@
+import hashlib
+import os
+import platform
+import shutil
+import stat
+import tarfile
+import tempfile
+import zipfile
+from dataclasses import dataclass
+from pathlib import Path
+
+import requests
+
+from halmos.logs import debug, error, info
+from halmos.utils import format_size
+
+# Define the cache directory for solvers
+SOLVER_CACHE_DIR = Path.home() / ".halmos" / "solvers"
+SOLVER_CACHE_DIR.mkdir(parents=True, exist_ok=True)
+
+YICES_BASE_URL = "https://github.com/SRI-CSL/yices2/releases/download/Yices-2.6.5"
```

Comment:
we're using 2.6.4 because of the performance degradation in the latest version:
https://github.com/a16z/halmos/blob/main/packages/solvers/Dockerfile#L33